src/components/PeopleTable.tsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect } from 'react';
+import cn from 'classnames';
+import { useParams } from 'react-router-dom';
+import { Person } from '../types';
+import { PeopleLink } from './PeopleLink';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+
+  function getPersonByName(name: string) {
+    return people.find(pers => pers.name === name);
+  }
+
+  useEffect(() => {
+    const selectedPerson = document.querySelector('.has-background-warning');
+
+    if (selectedPerson) {
+      selectedPerson.scrollIntoView({
+        block: 'center',
+        behavior: 'smooth',
+      });
+    } else {
+      window.scrollTo({ top: 0, behavior: 'smooth' });
+    }
+  }, [slug]);
```

Comment:
since you're mapping the whole table you can get selected element from (person), and you should almost never use `querySelector` in react, then this effect will be unnecessary